src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -109,9 +109,37 @@ internal override SyntaxNode ScopeDesignatorOpt
             get { return _scopeBinder.ScopeDesignator; }
         }
 
-        internal override uint RefEscapeScope => _refEscapeScope;
+        // From https://github.com/dotnet/csharplang/blob/main/proposals/low-level-struct-improvements.md:
+        //
+        // | Parameter or Local     | ref-safe-to-escape | safe-to-escape |
+        // |------------------------|--------------------|----------------|
+        // | Span<int> s            | current method     | calling method |
+        // | scoped Span<int> s     | current method     | current method |
+        // | ref Span<int> s        | calling method     | calling method |
+        // | scoped ref Span<int> s | current method     | calling method |
+        // | ref scoped Span<int> s | current method     | current method |
+
+        internal override uint RefEscapeScope
+        {
+            get
+            {
+                uint limit = _scope == DeclarationScope.Unscoped && _refKind != RefKind.None ?
+                    Binder.ExternalScope :
+                    Binder.TopLevelScope;
+                return Math.Max(_refEscapeScope, limit);
```

Comment:
>return Math.Max(_refEscapeScope, limit); [](http://example.com/codeflow?start=16&length=40)

Which part of the spec does this implement. I see it saying: "For locals these annotations concretely define the lifetimes **and override the lifetime that would be inferred via the initializer.**" But this doesn't look like overriding to me.

Reply:
Good catch, this does not agree the spec.

Jared, should the lifetime of the initializer be ignored if the local has a `scoped` modifier? I believe the only case where there is a difference is the _safe-to-escape_ of a `scoped ref` local where the initializer is `ref scoped`:
```csharp
ref struct R { }
class Program
{
    static R Refs(ref scoped R r3)
    {
        scoped ref R r43 = ref r3;
        return r43; // error?
    }
}
```

Reply:
I thought the local initializer would have an error in that scenario.